cli/internal/terraform/terraform/aws/main.tf

```diff
@@ -0,0 +1,301 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "3.4.1"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = var.region
+}
+
+locals {
+  uid              = random_id.uid.hex
+  name             = "${var.name}-${local.uid}"
+  tag              = "constellation-${local.uid}"
+  ports_node_range = "30000-32767"
+  ports_ssh        = "22"
+
+  ports_kubernetes   = "6443"
+  ports_bootstrapper = "9000"
+  ports_konnectivity = "8132"
+  ports_verify       = "30081"
+  ports_debugd       = "4000"
+
+  disk_size = 10
+
+  cidr_vpc_subnet_nodes    = "192.168.178.0/24"
+  cidr_vpc_subnet_internet = "192.168.0.0/24"
+}
+
+resource "random_id" "uid" {
+  byte_length = 4
+}
+
+resource "aws_vpc" "vpc" {
+  cidr_block = "192.168.0.0/16"
+  tags = {
+    Name = "${local.name}-vpc"
+  }
+}
+
+resource "aws_subnet" "private" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = local.cidr_vpc_subnet_nodes
+  availability_zone = var.zone
+  tags = {
+    Name = "${local.name}-subnet-nodes"
+  }
+}
+
+resource "aws_subnet" "public" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = local.cidr_vpc_subnet_internet
+  availability_zone = var.zone
+  tags = {
+    Name = "${local.name}-subnet-internet"
+  }
+}
+
+resource "aws_internet_gateway" "gw" {
+  vpc_id = aws_vpc.vpc.id
+
+  tags = {
+    Name = "${local.name}-internet-gateway"
+  }
+}
+
+resource "aws_nat_gateway" "gw" {
+  subnet_id     = aws_subnet.public.id
+  allocation_id = aws_eip.nat.id
+
+  tags = {
+    Name = "${local.name}-nat-gateway"
+  }
+}
+
+resource "aws_route_table" "private_nat" {
+  vpc_id = aws_vpc.vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_nat_gateway.gw.id
+  }
+
+  tags = {
+    Name = "${local.name}-nat-route"
+  }
+}
+
+resource "aws_route_table" "public_igw" {
+  vpc_id = aws_vpc.vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.gw.id
+  }
+
+  tags = {
+    Name = "${local.name}-nat-route"
+  }
+}
+
+resource "aws_route_table_association" "private-nat" {
+  subnet_id      = aws_subnet.private.id
+  route_table_id = aws_route_table.private_nat.id
+}
+
+resource "aws_route_table_association" "route_to_internet" {
+  subnet_id      = aws_subnet.public.id
+  route_table_id = aws_route_table.public_igw.id
+}
+
+resource "aws_eip" "lb" {
+  vpc = true
+}
+
+resource "aws_eip" "nat" {
+  vpc = true
+}
+
+resource "aws_lb" "front_end" {
+  name               = "${local.name}-loadbalancer"
+  internal           = false
+  load_balancer_type = "network"
+
+  subnet_mapping {
+    subnet_id     = aws_subnet.public.id
+    allocation_id = aws_eip.lb.id
+  }
+
+  tags = {
+    Name = "loadbalancer"
+  }
+
+  enable_cross_zone_load_balancing = true
+}
```

Comment:
Mh, not sure and not super familiar with Terraform conventions.

I think a module makes sense when reusing certain components. It could make sense moving the elastic IP & subnet logic could into a separate module (as for HA creating multiple of them make sense in the future). 

The load balancer only needs to be created once, but for HA would have multiple subnet mappings. I think it could remain here. though.

Does this sound like a good compromise for you?